### ISSUE TITLE
plugin_host-3.3 has exited unexpectedly

### ISSUE THREAD

**Issue by AFlowOfCode** [SublimeLinter/SublimeLinter-flake8#129]
Sublime Text build 4152
SublimeLinter v4.23.0

As soon as I install SublimeLinter-flake8, I get this message & all plugins stop working. On restart of ST the same thing happens immediately after opening. 

If I go to preferences > settings and add "SublimeLinter-flake8" to `ignored_packages` and then restart, all plugins (including SublimeLinter) work correctly again, because the flake8 linter is now ignored.

The debug console (<kbd>ctrl+`</kbd>) shows nothing special when I get this error message:
```
reloading python 3.3 plugin SublimeLinter-flake8.linter
error: plugin_host-3.3 has exited unexpectedly, some plugin functionality won't be available until Sublime Text has been restarted
```

This began after the ST update to build 4152. Noting that this plugin hasn't been updated for over a year, it's likely it hasn't kept up with some change or another.

Is this linter abandoned at this point? I hope not because it's very useful!

**Comment by kaste** [comment]
> Noting that this plugin hasn't been updated for over a year, it's likely it hasn't kept up with some change or another.

Not really. I use this plugin all the time. I'm on build 4156 right now.  Your 4152 is from August and I would think someone else would have reported this as well (flake8 is very popular).  

Do you have a manual override?  Maybe you hacked on SublimeLinter or this plugin at some time.

**Comment by AFlowOfCode** [comment]
4152 is the latest public release on the stable channel as far as I can tell ("Check for updates" says ST is up to date). 

I haven't ever customized anything & have been using this linter for a long time. A month or so ago plugin host started crashing every time I opened ST, which I ignored for a long time (using no plugins whatsoever). Today I uninstalled both plugins, reinstalled SublimeLinter on its own & everything was working fine. I then reinstalled this one and the problem began again. 

If there is a way to give you more detailed information, let me know. Otherwise maybe I should just update ST from some channel that has later builds than the stable channel, in the event that this is a problem with this particular build of ST itself.

**Comment by kaste** [comment]
You could start in Sublime in safe mode, and then just install PackageControl, SublimeLinter and this flake8 adapter here.  

Otherwise it could depend on the files you have open, because enabling/installing this plugin may trigger the linter already.  But if it crashes just on starting that would be before we actually *do* anything.  SublimeText would crash just by loading the plugin code which is very strange.

**Comment by AFlowOfCode** [comment]
You're right, the plugin host crash is dependent upon a `.py` file being open. If no files are open it doesn't crash, but as soon as a files is opened for which the linter would be triggered it crashes as described. This is true for both safe mode and regular.

**Comment by kaste** [comment]
>  the plugin host crash is dependent upon a .py file being open

Is this a specific python file or any will do?  Which OS and python/flake8 version are you on?

Anyway that will get tricky to debug.  If plugin_host is crashing that's like Python tumbles down. E.g. it is not enough to just throw an unhandled exception, that would just print the exception in the the console, and SublimeText will continue.  Tearing python down so it doesn't even throw anything anymore is rare.  

Can you downgrade or upgrade SublimeText?  https://www.sublimetext.com/dev  lists some versions.  E.g. https://download.sublimetext.com/sublime_text_build_4143_x64.zip  would be the link for the *previous* stable version on Windows. 

We maybe should ask on https://discord.com/channels/280102180189634562/280102180189634562 if we can get a stacktrace when plugin_hosts crashes.

**Comment by AFlowOfCode** [comment]
Today ST pushed an update (build 4166) and after I installed it and unignored the plugin it worked again. 

This affected any python file, as this is what would trigger the linter. Previously I did try upgrading to their latest dev build as well as downgrading to the previous stable version, which I thought surely should work as this problem was introduced after installing their last stable upgrade. However for reasons unknown to me it still had the same problem. 

I was unable to find the time to further debug the issue, but I wanted to post an update that whatever it was it now seems to have been fixed. That seems to indicate that it was completely related to ST itself. Thanks for your suggestions and assistance, and of course the great plugin.
